aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/GreengrassCliSteps.java

```diff
@@ -96,15 +97,20 @@ public void verifyLocalDeployment(String status, int timeout) throws Interrupted
 
     @VisibleForTesting
     String getLocalDeploymentStatus() {
-        String deploymentId = scenarioContext.get(LOCAL_DEPLOYMENT_ID);
-        String response = platform.commands().executeToString(CommandInput.builder()
-                .line(testContext.installRoot().resolve("bin").resolve("greengrass-cli").toString())
-                .addAllArgs(Arrays.asList("deployment", "status", "--deploymentId", deploymentId))
-                .build());
-        LOGGER.debug(String.format("deployment status response received for deployment ID %s is %s",
-                deploymentId, response));
-        String[] responseArray = response.split(":");
-        return responseArray[responseArray.length - 1].trim();
+        try {
+            String deploymentId = scenarioContext.get(LOCAL_DEPLOYMENT_ID);
+            String response = platform.commands().executeToString(CommandInput.builder()
+                    .line(testContext.installRoot().resolve("bin").resolve("greengrass-cli").toString())
+                    .addAllArgs(Arrays.asList("deployment", "status", "--deploymentId", deploymentId))
+                    .build());
+            LOGGER.debug(String.format("deployment status response received for deployment ID %s is %s",
+                    deploymentId, response));
+            String[] responseArray = response.split(":");
+            return responseArray[responseArray.length - 1].trim();
+        } catch (CommandExecutionException e) {
+            LOGGER.info("Exception occurred while getting the deployment status. Will try again", e);
+        }
+        return "";
```

Comment:
Where is it retrying? Once it enters catch block it will print the statement and exit.

Reply:
The command is retried in the following condition since `""` in  not one of the terminal statuses (FAILED/SUCCEEDED). 

https://github.com/aws-greengrass/aws-greengrass-testing/blob/0256d26dbc8bf986da412122a4e9b10539912d48/aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/GreengrassCliSteps.java#L94